### ISSUE TITLE
Colorbar appearance in gene trend plotting

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#495]
- [ ] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [x] New plotting function in `cr.pl`?
- [ ] Other?

I'm showing gene trends for 2 genes along 4 lineages in a grid, i.e. separate per lineage/gene and I'm coloring in absorption probabilities like so: 

```python
model = cr.ul.models.GAM(adata_raw, n_knots=n_knots)
    
fig_kwargs = {'model': model,
              'data_key': data_key, 
              'genes': ['Arx', 'Pdx1'], 
              'time_key': pseudotime, 
              'same_plot': False, 
              'hide_cells': False,
              'weight_threshold': weight_threshold, 
              'figsize': (20, 7)}

cr.pl.gene_trends(adata_raw, **fig_kwargs)
```
This gives the following plot:
![image](https://user-images.githubusercontent.com/24494650/108039373-4d2bd900-703c-11eb-8b25-e8ce66446fad.png)

I don't think it makes sense to show the colorbar twice - could I even color in separate things for the different genes here? I think just showing it once, say in the bottom right corner, would be much cleaner.

**Comment by Marius1311** [comment]
I just tried the same thing with discrete annotations, there, we only show the legend once, which I think makes sense (and looks much cleaner): 
![image](https://user-images.githubusercontent.com/24494650/108039738-ca574e00-703c-11eb-89c7-f87a8acd2797.png)

**Comment by michalk8** [comment]
> I don't think it makes sense to show the colorbar twice - could I even color in separate things for the different genes here? I think just showing it once, say in the bottom right corner, would be much cleaner.

There is 1 problem with this - if only 1 cbar is shown, the plots bellow will be slightly larger in width. I also think that
if you want to read out values in 1 row (gene), you can stay withing 1 row (and not go to the top) and cbar is much smaller than a legend, in terms of width, so it doesn't introduce that much clutter.
As an alternative, we could have 1 cbar spanning multiple rows, but for 5+ genes, it will get really large. I will try thinking about this and found a way so that if using only 1 cbar on the top right, the width of below plots stays the same.

**Comment by Marius1311** [comment]
I don't understand why this is different from showing only a single legend?

**Comment by michalk8** [comment]
I've given this some though, I think it's ok to keep it this way - can't find an elegant solution to this that wouldn't pollute the code + it's such a minor "nuissance". If you feel this should be solved in 1.3, feel free to reopen.

**Comment by Marius1311** [comment]
I agree, let's leave this as it is.
